client/src/send_and_confirm_transactions_in_parallel.rs

```diff
@@ -235,14 +247,12 @@ async fn sign_all_messages_and_send<T: Signers + ?Sized>(
         );
 
         if let Some(progress_bar) = progress_bar {
-            set_message_for_confirmed_transactions(
-                progress_bar,
-                context
-                    .num_confirmed_transactions
-                    .load(std::sync::atomic::Ordering::Relaxed),
-                context.total_transactions,
-                None,
+            let progress = progress_from_context_and_block_height(
+                context,
                 blockhashdata.last_valid_blockheight,
+            );
+            progress.set_message_for_confirmed_transactions(
+                progress_bar,
```

Comment:
not sure if this `progress_bar` really works in case if send all transactions async